sql/core/src/test/scala/org/apache/spark/sql/PlanStabilitySuite.scala

```diff
@@ -101,10 +101,13 @@ trait PlanStabilitySuite extends TPCDSBase with DisableAdaptiveExecutionSuite {
     new File(goldenFilePath, name)
   }
 
-  private def isApproved(dir: File, actualSimplifiedPlan: String): Boolean = {
-    val file = new File(dir, "simplified.txt")
-    val expected = FileUtils.readFileToString(file, StandardCharsets.UTF_8)
-    expected == actualSimplifiedPlan
+  private def isApproved(
+    dir: File, actualSimplifiedPlan: String, actualExplain: String): Boolean = {
+    val simplifiedFile = new File(dir, "simplified.txt")
+    val expectedSimplified = FileUtils.readFileToString(simplifiedFile, StandardCharsets.UTF_8)
+    val explainFile = new File(dir, "explain.txt")
+    val expectedExplain = FileUtils.readFileToString(explainFile, StandardCharsets.UTF_8)
+    expectedSimplified == actualSimplifiedPlan && expectedExplain == actualExplain
```

Comment:
Can we avoid the comparison between explains when `expectedSimplified != actualSimplifiedPlan`?